projects/igniteui-angular/src/lib/grids/grid-public-row.ts

```diff
@@ -549,6 +584,34 @@ export class IgxSummaryRow implements RowType {
         return this._summaries ? this._summaries : this.grid.dataView[this.index].summaries;
     }
 
+    /**
+     * Returns the view index calculated per the grid page.
+     */
+     public get viewIndex(): number {
+        if (this.grid.hasSummarizedColumns && this.grid.page > 0) {
+            if (this.grid instanceof IgxGridComponent) {
+                const grid = this.grid as IgxGridComponent;
+                if (grid.groupingExpressions.length) {
+                    const firstRowGroupInd = grid.groupingFlatResult.indexOf(this.grid.dataView[0]);
+                    const precedingGroupedRecs = grid.groupingFlatResult.filter((r, ind) =>
+                        this.grid.isGroupByRecord(r) && ind < firstRowGroupInd);
+                        return firstRowGroupInd + precedingGroupedRecs.filter(gr => this.grid.isExpandedGroup(gr)).length + this.index;
+                }
+            } else if (this.grid instanceof IgxTreeGridComponent) {
+                const grid = this.grid as IgxTreeGridComponent;
+                if (grid.summaryCalculationMode !== GridSummaryCalculationMode.rootLevelOnly) {
+                    const firstRowIndex = grid.processedExpandedFlatData.indexOf(grid.dataView[0].data);

```

Comment:
Same as above applies to the TreeGrid as well, while using paging the returned index is incorrect. I have 7 rows (indexes from 0-6) on the first page (5 regular and 2 summary rows) and the viewIndex of the first row of page 2 should be with value 7, not 8